packages/ui/src/components/Button/SButton.vue

```diff
@@ -55,15 +70,15 @@ const font = computed(() => {
   <button
     :type="nativeType"
     :class="[
-      's-button',
-      `s-button_type_${definitelyType}`,
-      `s-button_size_${definitelySize}`,
-      `s-button_icon-position_${definitelyIconPosition}`,
+      bem.block,
+      bem[`block_type_${definitelyType}`],
+      bem[`block_size_${definitelySize}`],
+      bem[`block_iconPosition_${definitelyIconPosition}`],
```

Comment:
As far as I understand, there will be error caused camel case transforming, if there are constant with hyphen like `left-bottom` or smth like this.

Reply:
Hmm, you are right. TypeScript will catch it, but it doesn't solve the issue.

Reply:
@Ctepan,

what do you think about this kind of solution?

```ts
defineBem('s-button')
  .mod('accent-primary', { kebab: true, camel: false })
  .mod('type', ['accent-primary', 'accent-secondary'] as const, { kebab: true })
  .build()
```

Or maybe it even makes sense to preserve all passed element and modifiers key/value as-is for this case, but also provide `camelCase` for convenient access via JavaScript:

```ts
const bem = defineBem('s-button')
  .mod('accent-primary')
  .mod('ty-pe', ['accent-primary', 'accent-secondary'] as const)
  .build('classic', { preserve: true })
  
bem.block_tyPe_accentPrimary
bem['block_ty-pe_accent-primary']
```

Reply:
The second variant seems better, as for me. May be even turned on by default